pkg/controller/ec2/vpccidrblock/controller.go

```diff
@@ -121,17 +121,17 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	observed := response.Vpcs[0]
 
 	currentStatusCode, err := ec2.FindVPCCIDRBlockStatus(meta.GetExternalName(cr), observed)
-	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(ec2.IsCIDRNotFound, err), errDescribe)
+
+	// vpccidrblock is in status disassociated after deletion
+	if currentStatusCode == awsec2.VpcCidrBlockStateCodeDisassociated {
+		return managed.ExternalObservation{}, errors.Wrap(err, errDescribe)
 	}
```

Comment:
@haarchri it appears we are now not handling the error from `ec2.FindVPCCIDRBlockStatus` unless the status code is `awsec2.VpcCidrBlockStateCodeDisassociated`, which I don't think will work. Feel free to break this out into a separate PR so we can discuss without blocking the PRs which depend on this package change 👍🏻

Reply:
done - i will open a followup pr for this